packages/api/tests/endpoints/test-granules.js

```diff
@@ -843,40 +852,132 @@ test.serial('move a granule with no .cmr.xml file', async (t) => {
-          key: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.jpg`,
+          regex: '.*.jpg$',
+          bucket,
+          filepath: destinationFilepath,
         },
       ];
 
-      await granuleModel.create(newGranule);
-
-      await Promise.all(
-        newGranule.files.map(
-          (file) =>
-            s3PutObject({
-              Bucket: file.bucket,
-              Key: file.key,
-              Body: 'test data',
-            })
-        )
+      const response = await request(app)
+        .put(`/granules/${newGranule.granuleId}`)
+        .set('Accept', 'application/json')
+        .set('Authorization', `Bearer ${jwtAuthToken}`)
+        .send({
+          action: 'move',
+          destinations,
+        })
+        .expect(200);
+
+      const body = response.body;
+      t.is(body.status, 'SUCCESS');
+      t.is(body.action, 'move');
+
+      // Validate S3 Objects are where they should be
+      const bucketObjects = await s3().listObjects({
+        Bucket: bucket,
+        Prefix: destinationFilepath,
+      }).promise();
+
+      t.is(bucketObjects.Contents.length, 2);
+      bucketObjects.Contents.forEach((item) => {
+        t.is(item.Key.indexOf(`${destinationFilepath}/${granuleFileName}`), 0);
+      });
+
+      const thirdBucketObjects = await s3().listObjects({
+        Bucket: thirdBucket,
+        Prefix: destinationFilepath,
+      }).promise();
+
+      t.is(thirdBucketObjects.Contents.length, 1);
+      t.is(thirdBucketObjects.Contents[0].Key, `${destinationFilepath}/${granuleFileName}.md`);
+
+      // check the granule in dynamoDb is updated and files are replaced
+      const updatedGranule = await granuleModel.get({ granuleId: newGranule.granuleId });
+
+      updatedGranule.files.forEach((file) => {
+        t.true(file.key.startsWith(`${destinationFilepath}/${granuleFileName}`));
+        const destination = destinations.find((dest) => file.fileName.match(dest.regex));
+        t.is(destination.bucket, file.bucket);
+      });
+
+      // check the granule in postgres is updated
+      const pgFiles = await getPostgresFilesInOrder(
+        t.context.knex,
+        newGranule,
+        filePgModel,
+        postgresGranuleCumulusId
       );
 
-      const destinationFilepath = `${process.env.stackName}/granules_moved`;
+      t.is(pgFiles.length, 3);
+
+      for (let i = 0; i < pgFiles.length; i += 1) {
+        const destination = destinations.find((dest) => pgFiles[i].file_name.match(dest.regex));
+        t.is(destination.bucket, pgFiles[i].bucket);
+        t.like(pgFiles[i], {
+          ...omit(newGranule.files[i], ['fileName', 'size']),
+          key: `${destinationFilepath}/${newGranule.files[i].fileName}`,
+          bucket: destination.bucket,
+          file_name: newGranule.files[i].fileName,
+        });
+      }
+    }
+  );
+});
+
+test.serial('a granule that fails to move a file correctly reports the granule status', async (t) => {
```

Comment:
Where in here to do we verify the granule status? I think we mean that when a granule fails to move, the granule file records are still updated?

Reply:
Yeah, that's not a good title.  How about: 55b9d5b7f